File: Question1Solution.java
import java.util.HashMap;

public class FirstRepeatedCharacter {
    public static char firstRepeatedCharacter(String str) {
        HashMap<Character, Integer> charCount = new HashMap<>();
        for (char ch : str.toCharArray()) {
            charCount.put(ch, charCount.getOrDefault(ch, 0) + 1);
            if (charCount.get(ch) > 1) {
                return ch;
            }
        }
        return '\0';
    }

    public static void main(String[] args) {
        String str = "hello";
        char result = firstRepeatedCharacter(str);
        if (result != '\0') {
            System.out.println("First repeated character: " + result);
        } else {
            System.out.println("No repeated characters found.");
        }
    }
}

File: Question1Solution.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

char firstRepeatedCharacter(const string &str) {
    unordered_map<char, int> charCount;
    for (char ch : str) {
        charCount[ch]++;
        if (charCount[ch] > 1) {
            return ch;
        }
    }
    return '\0';
}

int main() {
    string str = "hello";
    char result = firstRepeatedCharacter(str);
    if (result != '\0') {
        cout << "First repeated character: " << result << endl;
    } else {
        cout << "No repeated characters found." << endl;
    }
    return 0;
}
